UnsubscribeFunction/UnsubscribeFunction.py

```diff
@@ -0,0 +1,97 @@
+import os
+import json
+import boto3
+from botocore.vendored import requests
+from boto3.dynamodb.conditions import Key
+from botocore.exceptions import ClientError
+
+dynamo_client = boto3.resource('dynamodb')
+table = boto3.resource('dynamodb').Table(os.environ['TABLE_NAME'])
+
+# Unsubscribe a user from the given HSK level or all levels.
+def lambda_handler(event, context):
+
+    body = json.loads(event["body"])
+
+    # Extract relevant user details
+    email_address = body['email']
+    hsk_level = body['level']
+
+    # Call Dynamo to check if user is subscribed to the given level
+    response, contact_found_count = find_contact(email_address, hsk_level)
+    print("Found contacts: ", contact_found_count)
+
+    unsubscribe_user(response, contact_found_count)
+
+    return {
+        'statusCode': 200,
+        'headers': {
+            'Access-Control-Allow-Methods': 'POST,OPTIONS',
+            'Access-Control-Allow-Origin': '*',
+        },
+        'body': '{"success" : true}'
+      }
+
+def find_contact(email_address, hsk_level):
+
+    keys = []
+
+    # If unsubscribing from all lists, get item from Dynamo by subscriber email and each HSK level list
+    if hsk_level == "all":
+
+      # Generate a list of all HSK level keys
+      for level_list in range(0,5):
+        keys.append({
+          'ListId': str(level_list+1),
+          'SubscriberEmail': email_address
+        })
+
+    # If unsubscribing from a single list, get item from Dynamo by subscriber email and the given HSK level list
+    else:
+      keys.append({
+          'ListId': hsk_level,
+          'SubscriberEmail': email_address
+        })
+
+    # Batch get item from Dynamo
+    try:
+      response = dynamo_client.batch_get_item(
+        RequestItems={
+          table.name : {
+            'Keys' : keys
+          }
+        }
+      )
+    except ClientError as e:
+      print(e.response['Error']['Message'])
+    else:
+      print(response)
+      contact_found_count = len(response["Responses"][table.name])
+
+    return response, contact_found_count
```

Comment:
I think it might be a little cleaner if this function just returned `return response["Responses"][table.name]`. That way the `unsubscribe_user` function can take a single parameter of `subscriptions` and instead of passing the count around between functions you can just use `len(subscriptions)`